tests/sles4sap/publiccloud/qesap_ansible.pm

```diff
@@ -35,7 +36,47 @@ sub run {
     # skip ansible deployment in case of reusing infrastructure
     unless (get_var('QESAP_DEPLOYMENT_IMPORT')) {
         my @ret = qesap_execute(cmd => 'ansible', timeout => 3600, verbose => 1);
-        die("Ansible deploymend FAILED. Check 'qesap*' logs for details.") if $ret[0] > 0;
+        if ($ret[0]) {
```

Comment:
I am thinking how to break down the nested structure here.
Maybe you could make a separate function from the whole if statement or from some of the underlying statements.

Reply:
I moved the nested code out of `if (qesap_file_find_string())` to fix the nested issue.